tests/v2/validation/provisioning/rke2/etcd_only_backup_restore_test.go

```diff
@@ -0,0 +1,289 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	provisioningV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/ingresses"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kubeProvisioning "github.com/rancher/rancher/tests/framework/clients/provisioning"
+	networkingv1 "k8s.io/api/networking/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+type RKE2EtcdOnlySnapshotRestore struct {
+	suite.Suite
+	session            *session.Session
+	client             *rancher.Client
+	ns                 string
+	kubernetesVersions []string
+	cnis               []string
+	providers          []string
+	nodesAndRoles      []machinepools.NodeRoles
+}
+
+func (r *RKE2EtcdOnlySnapshotRestore) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2EtcdOnlySnapshotRestore) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	r.kubernetesVersions = clustersConfig.RKE2KubernetesVersions
+	r.cnis = clustersConfig.CNIs
+	r.providers = clustersConfig.Providers
+	r.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+}
+
+func (r *RKE2EtcdOnlySnapshotRestore) EtcdSnapshotRestore(provider *Provider) {
+	initialK8sVersion := r.kubernetesVersions[0]
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := r.client.WithSession(subSession)
+	require.NoError(r.T(), err)
+
+	logrus.Infof("creating kube provisioning client.............")
+	kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+	require.NoError(r.T(), err)
+	logrus.Infof("kube provisioning client created.............")
+
+	clusterName := provisioning.AppendRandomString(provider.Name)
```

Comment:
I noticed this when reviewing the backport. This should actually be using `namegen` as the backport is using. This is due to Corral changes that recently were introduced.

Reply:
Yes Vivek please rebase your branch, we have had a few incidents with breaking tests because we neglected to rebase our branches for the most recent code.

Reply:
Updated as per comment. Changes are in the recent commit.